tests/execution/test-required-fields.md

```diff
@@ -0,0 +1,126 @@
+# Test API
+
+```graphql @config
+schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com") {
+  query: Query
+}
+
+type Query {
+  users: [User] @http(path: "/users")
+  posts: [Post] @http(path: "/posts")
+  post: Post! @http(path: "/posts/12")
+  foo: [Foo]! @http(path: "/foos")
+  fooInner: [Foo!] @http(path: "/foos-inner")
```

Comment:
```suggestion
  foosInner: [Foo!] @http(path: "/foos-inner")
```

Reply:
not sure, check the spec. Empty array is a valid array and since the mock returns empty list everything is fine.

it should throw an error in case only if inner type is marked as non-null also and some of the return values in list are resolved to null